src/librustc_errors/diagnostic.rs

```diff
@@ -315,10 +319,35 @@ impl Diagnostic {
         self
     }
 
+    pub fn span_suggestion_verbose(
+        &mut self,
+        sp: Span,
+        msg: &str,
+        suggestion: String,
+        applicability: Applicability,
+    ) -> &mut Self {
+        self.suggestions.push(CodeSuggestion {
+            substitutions: vec![Substitution {
+                parts: vec![SubstitutionPart {
+                    snippet: suggestion,
+                    span: sp,
+                }],
+            }],
+            msg: msg.to_owned(),
+            style: SuggestionStyle::ShowAlways,
```

Comment:
Save for `style`, this logic is repeated in `span_suggestion`, `span_suggestion_short`, `span_suggestion_hidden`, and `tool_only_span_suggestion`. Can you please de-dup that logic with e.g. `fn span_suggestion_with_style`?